clients/python/moondream/cloud_vl.py

```
@@ -0,0 +1,161 @@
+        image: Union[Image.Image, EncodedImage],
+        length: str = "normal",
+        stream: bool = False,
+        settings: Optional[SamplingSettings] = None,
+    ) -> CaptionOutput:
+        encoded_image = self.encode_image(image)
+
+        payload = {
+            "image_url": encoded_image.image_url,
+            "length": length,
+            "stream": stream,
+        }
+
+        data = json.dumps(payload).encode("utf-8")
+        headers = {"X-Moondream-Auth": self.api_key, "Content-Type": "application/json"}
+        req = urllib.request.Request(
+            f"{self.api_url}/caption",
+            data=data,
+            headers=headers,
+        )
+
+        def generator():
+            for chunk in self._stream_response(req):
+                yield chunk
+
+        if stream:
+            return {"caption": generator()}
+
+        with urllib.request.urlopen(req) as response:
+            result = json.loads(response.read().decode("utf-8"))
+            return {"caption": result.get("caption", "")}
+
+    def query(
+        self,
+        image: Union[Image.Image, EncodedImage],
+        question: str,
+        stream: bool = False,
+        settings: Optional[SamplingSettings] = None,
+    ) -> QueryOutput:
+        encoded_image = self.encode_image(image)
+
+        payload = {
+            "image_url": encoded_image.image_url,
+            "question": question,
+            "stream": stream,
+        }
+
+        data = json.dumps(payload).encode("utf-8")
+        headers = {"X-Moondream-Auth": self.api_key, "Content-Type": "application/json"}
+        req = urllib.request.Request(
+            f"{self.api_url}/query",
+            data=data,
+            headers=headers,
+        )
+
+        if stream:
+            return {"answer": self._stream_response(req)}
+
+        with urllib.request.urlopen(req) as response:
+            result = json.loads(response.read().decode("utf-8"))
+            return {"answer": result.get("answer", "")}
```

should throw exception instead of returning empty string